src/resolver_fallback.rs

```diff
@@ -0,0 +1,67 @@
+use hyper_util::client::legacy::connect::dns::GaiResolver;
+use reqwest::dns::{Addrs, Name, Resolve, Resolving};
+use std::net::SocketAddr;
+use std::str::FromStr;
+use std::sync::Arc;
+use tower_service::Service;
+
+pub trait ResolverFallbackCache: Send + Sync {
+    fn get(&self, name: &str) -> Vec<SocketAddr>;
+    fn set(&self, name: &str, addr: &[SocketAddr]);
```

Comment:
I'm surprised that these don't need to be async.

Reply:
Maybe I'm wrong. We'll see. But interactions with the `ClientState` usually don't need async.